go/vt/topo/etcd2topo/watch.go

```diff
@@ -51,7 +51,7 @@ func (s *Server) Watch(ctx context.Context, filePath string) (*topo.WatchData, <
 	}
 	wd := &topo.WatchData{
 		Contents: initial.Kvs[0].Value,
-		Version:  EtcdVersion(initial.Kvs[0].ModRevision),
+		Version:  EtcdVersion(initial.Kvs[0].Version),
```

Comment:
What is the difference between these two?

Reply:
Revision, ModRevision, and Version are similar but distinct things:
  - https://github.com/etcd-io/etcd/issues/6518
  - https://etcd.io/docs/v3.5/learning/data_model/

I'll try to summarize:
  - Revision: a logical clock used to track changes to the etcd keyspace
  - Version: a logical clock used to track the changes to key since it was created
  - ModRevision: the etcd keyspace revision number for the most recent update to a key (the latest version)

What I tried to do is align the Vitess comments and code so that we're matching etcd for which one of these to use. As you can see in this example here, we were using Version in Vitess code and setting it to the ModRevision value from etcd. When I first started the investigate I thought that, aside from this being odd, it could have perhaps led us to miss/skip an update to the key in the watch.

Reply:
Good one